utils/wasm-gen/src/tests.rs

```diff
@@ -47,6 +47,40 @@ use std::{mem, num::NonZeroUsize};
 
 const UNSTRUCTURED_SIZE: usize = 1_000_000;
 
+#[test]
+fn inject_critical_gas_limit_works() {
```

Comment:
please open an issue with P2 or P3 priority to introduce test that actually limiter's logic works as expected

Reply:
#3502